src/hyp_map.rs

```diff
@@ -410,6 +396,20 @@ impl HypMap {
         self.private_regions.iter()
     }
 
+    /// Returns the virtual address of U-mode mapping slot `slot`.
+    pub fn umode_slot_va(slot: UmodeSlotId) -> PageAddr<SupervisorVirt> {
+        let slot_num = match slot {
```

Comment:
Alternatively:

let slot_num = UmodeSlotId as usize;

(and if you want to be extra sure, give the enum cases explicit values)

Reply:
Aware. I'd rather keep it like this to be explicit on their mapping. Giving explicit values in a enum as I've done for other types would just make it harder to understand where the address go.

In general in case SlotId grows as a type this should become a function associated to the type (unsure of the rust slang for this), but for now I like this little functions because on reading tells you exactly where the VA goes.